protocols/DNSADDR.md

```diff
@@ -0,0 +1,38 @@
+# `dnsaddr`
```

Comment:
@alanshaw 

This should probably explain suffix matching: Given the address `/dnsaddr/bootstrap.libp2p.io/SOMETHING`, this address will only resolve to multiaddrs that end with SOMETHING.